src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -1204,7 +1216,7 @@ bool CanContinueBatch(List<string> idsToCheck, TimeSpan elapsed, JsonOperationCo
             MinimumRevisionsToKeep = 0
         };
 
-        private long EnforceConfigurationFor(DocumentsOperationContext context, string id)
+        private long EnforceConfigurationFor(DocumentsOperationContext context, string id, List<string> hasMoreIdsList)
```

Comment:
This adds unnecessary complexity to a very rare case. Let's just track in the command if there is more work (bool). If there is, just fire command with the same set of Ids.